app/controllers/security_ruleset_controller.rb

```diff
@@ -0,0 +1,41 @@
+
+class SecurityRulesetController < ApplicationController
+  include RulesetValidations
+
+  before_action :validate_file_upload!, only: [:create]
+  before_action :validate_ruleset_name!, only: [:new]
+  before_action :clear_errors
+
+  def new
+  end
+
+  def create
+    crunch_json = JSON.parse(crunch42_data)
+    @score = crunch_json['score']
+    @openapi = crunch_json['openapiState']
+    @counter = crunch_json['issueCounter']
+    @criticality = crunch_json['criticality']
+    @issues = crunch_json['data']['issues'].map do |issue|
+      lines = issue[1]['issues'].map { |occurrence| occurrence['pointer'] }
+      [issue[1]['criticality'],issue[1]['description'], lines]
+    end
+
+    render 'show'
```

Comment:
Rendering `show` here is the reason flash is being problematic.

In a normal rails controller the submission from the form on the `new` view would be sent to the create action. If all is OK the create action would **redirect** to the `show` action, perhaps with a flash message. On failure the original form would be **rendered** with an error object passed to the form.

By using flash messages to display error messages on a render view rather than a redirect, the flash system isn't behaving as it is designed to do. Therefore, the `clear_errors` method is needed to "fix" this problem.

If the behaviour was broken down into two actions:

- **create** could be dedicated with checking the validity of the input and handling any errors with that input
- **show** could be dedicated to manipulating the data to display for the use.

The issue then becomes how to persist the data between the two action. The simplest solution is to persist it in the database via an ActiveRecord object. That's how rails expects things to work and you get all the rails methods (validations for example) to make things easier.

If storing data in a postgres database is problematic there other options. Use ActiveModel and a memcache for example. You might even get away with using session (especially if session is configured to use a memcache store.

Reply:
We do have postgres, and we do have a plan for setting out a model we can use which will make life a bit easier, however that would be further down the pipeline. Within this pr I don't want to implement this as its not needed right now.